naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2402,17 +2448,24 @@ impl Parser {
         // read item
         let start = lexer.start_byte_offset();
         let kind = match lexer.next() {
-            (Token::Separator(';'), _) => None,
+            (Token::Separator(';'), _) => {
+                ensure_no_diag_attrs("`struct`s", diagnostic_filters)?;
```

Comment:
```suggestion
                ensure_no_diag_attrs("`;`s", diagnostic_filters)?;
```

Reply:
Ah, yikes, sorry, trying to operate too quickly. 😓 Fixed to say `semicolons` and added grammar tweaks to the error messages being added with 9141923e5b15dbec2e8b8cacc96c92b0ba170ee3.

This diagnostic is weird, because there might be a case where somebody _just_ did `@diagnostic(…);`. It'd be nice to offer advice, indicating that they may have wanted a directive instead.

This is a bit complex to handle in the face of other parsed attributes, so I won't do that here, but I'll file a follow-up issue: <https://github.com/gfx-rs/wgpu/issues/6536>